File: app/src/main/java/com/app/missednotificationsreminder/di/Injector.java
package com.app.missednotificationsreminder.di;

import android.content.Context;

import dagger.ObjectGraph;

/**
 * Utility class which allows to obtain {@link ObjectGraph} related to {@link Context}
 */
public final class Injector {
    private static final String INJECTOR_SERVICE = Injector.class.getName();

    /**
     * Obtain the {@link ObjectGraph} related to the specified context.
     *
     * @param context
     * @return
     */
    @SuppressWarnings({"ResourceType", "WrongConstant"}) // Explicitly doing a custom service.
    public static ObjectGraph obtain(Context context) {
        return (ObjectGraph) context.getSystemService(INJECTOR_SERVICE);
    }

    public static boolean matchesService(String name) {
        return INJECTOR_SERVICE.equals(name);
    }

    private Injector() {
        throw new AssertionError("No instances.");
    }
}

File: app/src/main/java/com/app/missednotificationsreminder/di/Injector.kt
package com.app.missednotificationsreminder.di

import android.content.Context
import dagger.android.AndroidInjector
import dagger.android.support.DaggerApplication

/**
 * Utility class which allows to obtain [AndroidInjector] related to [Context]
 */
class Injector private constructor() {
    companion object {
        /**
         * Obtain the [AndroidInjector] related to the specified context.
         *
         * @param context
         * @return
         */
        // Explicitly doing a custom service.
        @JvmStatic
        fun obtain(context: Context): AndroidInjector<Any>? {
            val app = context.applicationContext;
            return if (app is DaggerApplication) app.androidInjector() else null
        }
    }

    init {
        throw AssertionError("No instances.")
    }
}